modules/apps/fragment/fragment-entry-processor/fragment-entry-processor-freemarker/src/main/java/com/liferay/fragment/entry/processor/freemarker/FreeMarkerFragmentEntryProcessor.java

```diff
@@ -78,7 +79,7 @@ public String processFragmentEntryLinkHTML(
 			freeMarkerFragmentEntryProcessorConfiguration =
 				_configurationProvider.getCompanyConfiguration(
 					FreeMarkerFragmentEntryProcessorConfiguration.class,
-					fragmentEntryLink.getCompanyId());
+					CompanyThreadLocal.getCompanyId());
```

Comment:
This doesn't seem right. How does the fragmentEntryLink from a different company get here? We should figure out where the fragment is fetched and make sure it's fetching the one with the correct companyId.